index.js

```diff
@@ -2,19 +2,35 @@
 const path = require('path');
 const locatePath = require('locate-path');
 
+function isLastDirEqualTo(dir, compareDir) {
+	if (!compareDir) {
+		return false;
+	}
+
+	const splittedPath = dir.split('/');
+	const lastFolder = splittedPath[splittedPath.length - 1];
+
+	if (lastFolder === compareDir) {
+		return true;
+	}
+
+	return false;
+}
+
 module.exports = (filename, opts) => {
 	opts = opts || {};
 
 	const startDir = path.resolve(opts.cwd || '');
 	const root = path.parse(startDir).root;
-
 	const filenames = [].concat(filename);
 
 	return new Promise(resolve => {
 		(function find(dir) {
 			locatePath(filenames, {cwd: dir}).then(file => {
 				if (file) {
 					resolve(path.join(dir, file));
+				} else if (isLastDirEqualTo(dir, opts.stopDir)) {
```

Comment:
Conceptually, this is the same as the next `else if`; the point is we are stopping and returning `null` because we have reached the end. I would prefer to combine these in the form of `else if (foo || bar)`.

Reply:
You need to `path.resolve(opts.stopDir)`, otherwise comparisons against it will fail if the user passes a relative path like `../../`.